feature/experimental/gribi/ate_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,565 @@
+		t.Fatalf("Controller %q did not become switchover-ready before test.", primaryBeforeSwitch)
+	}
+
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	switchoverRequest := &spb.SwitchControlProcessorRequest{
+		ControlProcessor: &tpb.Path{
+			Elem: []*tpb.PathElem{{Name: secondaryBeforeSwitch}},
+		},
+	}
+	t.Logf("switchoverRequest: %v", switchoverRequest)
+
+	entriesBefore := checkNIHasNEntries(ctx, client, *deviations.DefaultNetworkInstance, t)
+
+	// Concurrently run switchover and gribi route flush
+	t.Log("Execute gRIBi flush and master switchover concurrently")
+	go func(msg string) {
+		if _, err := gribi.Flush(client, eID, *deviations.DefaultNetworkInstance); err != nil {
+			t.Logf("Unexpected error from flush, got: %v", err)
+		}
+	}("gRIBi Flush")
+
+	go func(msg string) {
+		switchoverResponse, err := gnoiClient.System().SwitchControlProcessor(context.Background(), switchoverRequest)
+		if err != nil {
+			t.Logf("Failed to perform control processor switchover with unexpected err: %v", err)
+		}
+		t.Logf("gnoiClient.System().SwitchControlProcessor() response: %v, err: %v", switchoverResponse, err)
+	}("Master Switchover")
+
+	t.Log("Switchover/Flush is completed, validate switchoverStatus now")
+
+	primaryAfterSwitch := validateSwitchoverStatus(t, dut, secondaryBeforeSwitch)
+
+	validateSwitchoverTelemetry(t, dut, primaryAfterSwitch)
+
+	// Following reconnection of the gRIBI client to a new master supervisor,
+	// validate if partially deleted entries of IPBlock1  are not present in the FIB
+	// using a get RPC.
+	// Connect gRIBI client to DUT referred to as gRIBI - using PRESERVE persistence and
+	// SINGLE_PRIMARY mode, with FIB ACK requested. Specify gRIBI as the leader.
+	// Check vars for WithInitialElectionID
+	t.Log("Reconnect gRIBi client after switchover on new master")
+	client.Connection().WithStub(gribic).WithPersistence().WithInitialElectionID(eID.Low, eID.High).
+		WithFIBACK().WithRedundancyMode(fluent.ElectedPrimaryClient)
+	client.Start(ctx, t)
+	defer client.Stop(t)
+
+	defer func() {
+		// Flush all entries after test.
+		if err := gribi.FlushAll(client); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// Reconnect gribi client
+	client.StartSending(ctx, t)
+
+	if err := awaitTimeout(ctx, client, t, time.Minute); err != nil {
+		t.Fatalf("Await got error during session negotiation for clientA: %v", err)
+	}
+
+	t.Log("Compare route entries after switchover")
+	entriesAfter := checkNIHasNEntries(ctx, client, *deviations.DefaultNetworkInstance, t)
+
+	if entriesAfter >= entriesBefore {
+		t.Error("After switchover, on new master seeing unexpected number of route entries")
+		t.Errorf("Network instance has %d entries before switchover, found after switchover: %d", entriesBefore, entriesAfter)
+	}
+
+	// TODO: Check for coredumps in the DUT and validate that none are present post failover
+
+	t.Log("Re-inject routes from IPBlock1 in default VRF with NHGID: #1.")
+	time.Sleep(3 * time.Minute)
```

Comment:
Please add a comment to justify why three minutes sleep is required here?

Reply:
updated comment, please check